.github/actions/setup-binary/action.yml

```
@@ -0,0 +1,135 @@
+
+inputs:
+  download_url:
+    description: 'The download URL for the binary. This must a valid go-getter URL. See https://github.com/hashicorp/go-getter#url-format'
+    required: true
+  checksum:
+    description: 'The checksum for the downloaded artifact. See https://github.com/hashicorp/go-getter#checksumming'
+    required: false
+  install_path:
+    description: 'The path on disk in which to install the binary (defaults to the current directory). This includes the folder and binary name (e.g. /path/to/my/binary). This action does not modify $PATH; if you want the binary to be accessible, make sure the parent folder is in $PATH.'
+    default: '.'
+    required: false
+  cache_key:
+    description: 'The cache key to use when caching the downloaded binary. If this value is empty, caching will be disabled.'
+    required: false
+  add_to_path:
+    description: 'Add the install_path to the $PATH variable.'
+    default: 'false'
+    required: false
+
+runs:
+  using: 'composite'
+  steps:
+    - name: 'Validate inputs'
+      shell: 'bash'
+      env:
+        DOWNLOAD_URL: '${{ inputs.download_url }}'
+      run: |
+        # Validate inputs
+        if [[ -z "${DOWNLOAD_URL}" ]]; then
+            echo "::error ::download_url is a required input"
+            exit 1
+        fi
+
+    - name: 'Restore cache binary'
+      id: 'cache-binary'
+      if: |
+        inputs.cache_key != ''
+      uses: 'actions/cache/restore@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8' # ratchet:actions/cache/restore@v3
+      with:
+        path: '${{ inputs.install_path }}'
+        key: 'setup-binary-${{ inputs.cache_key }}'
+
+    - name: 'Install hashicorp/go-getter'
+      if: |
+        steps.cache-guardian.outputs.cache-hit != 'true'
+      shell: 'bash'
+      env:
+        VERSION: '1.7.1'
+      run: |-
+        # Install hashicorp/go-getter
+
+        EXISTS=$(which go-getter)
+        echo "${EXISTS}"
+
+        if [[ ! -z "${EXISTS}" ]]; then
+          echo "go-getter is already installed"
+          exit 0
+        fi
+
+        if [[ "${RUNNER_OS}" == "Linux" ]]
+        then
+          TEMP_DIR="${RUNNER_TEMP}/go-getter"
+          RELEASE_URL="https://github.com/hashicorp/go-getter/releases/download/v${VERSION}/go-getter_${VERSION}_linux_amd64.zip"
+        elif [[ "${RUNNER_OS}" == "macOS" ]]
+        then
+          TEMP_DIR="${RUNNER_TEMP}/go-getter"
+          RELEASE_URL="https://github.com/hashicorp/go-getter/releases/download/v${VERSION}/go-getter_${VERSION}_darwin_amd64.zip"
+        elif [[ "${RUNNER_OS}" == "Windows" ]]
+        then
+          TEMP_DIR="${RUNNER_TEMP}\go-getter"
+          RELEASE_URL="https://github.com/hashicorp/go-getter/releases/download/v${VERSION}/go-getter_${VERSION}_windows_amd64.zip"
+        else
+          echo "::error ::Unsupported operating system ${RUNNER_OS}"
+          exit 1
+        fi
+
+        mkdir -p "${TEMP_DIR}" && cd "${TEMP_DIR}"
+        curl -L "${RELEASE_URL}" -o go-getter.zip
```

```suggestion
        curl -sfL "${RELEASE_URL}" -o go-getter.zip
```

-s will hide output, -f will fail on non-200 responses (after redirects)